source/kernel/src/lib.rs

```diff
@@ -331,31 +245,15 @@ impl Kernel {
     where
         F: Future + 'static,
     {
-        let task = self.new_task(fut);
-        let mut guard = self
-            .heap()
-            .lock()
-            .map_err(|_| "kernel heap already locked")?;
-        let task_box = guard
-            .alloc_box(task)
-            .map_err(|_| "could not allocate task storage")?;
-        Ok(self.spawn_allocated(task_box))
-    }
-
-    pub fn new_task<F>(&'static self, fut: F) -> Task<F>
-    where
-        F: Future + 'static,
-    {
-        Task(MaitakeTask::new(fut))
+        Ok(self.inner.scheduler.spawn(fut))
     }
 
     pub async fn spawn<F>(&'static self, fut: F) -> JoinHandle<F::Output>
     where
         F: Future + 'static,
     {
-        let task = Task(MaitakeTask::new(fut));
-        let atask = self.heap().allocate(task).await;
-        self.spawn_allocated(atask)
+        // TODO: allocator await?
```

Comment:
we could do this by allocating the box ourselves and using `maitake`'s `spawn_allocated` with `BoxStorage`, which is provided by maitake out of the ... box. :)

Reply:
I tried doing that, and it caused a lot of problems, but that was very mid-switchover, maybe it'd be nicer now.